src/test/java/org/prebid/server/bidder/taboola/TaboolaBidderTest.java

```diff
@@ -0,0 +1,326 @@
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(BidderBid::getBid)
+                .containsOnly(Bid.builder()
+                        .impid("123")
+                        .build());
+        assertThat(result.getValue())
+                .extracting(BidderBid::getType)
+                .containsExactly(BidType.banner);
+    }
+
+    @Test
+    public void makeBidsShouldReturnValidBidResponseWithBannerTypeWhenImpIncludeNativeAndBanner()
+            throws JsonProcessingException {
+        // given
+        final BidRequest bidRequest = givenBidRequest(impCustomizer -> impCustomizer
+                .xNative(Native.builder().build())
+                .banner(Banner.builder().build()));
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                bidRequest,
+                mapper.writeValueAsString(
+                        givenBidResponse(identity())));
+
+        // when
+        final Result<List<BidderBid>> result = taboolaBidder.makeBids(httpCall, bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(BidderBid::getBid)
+                .containsOnly(Bid.builder()
+                        .impid("123")
+                        .build());
+        assertThat(result.getValue())
+                .extracting(BidderBid::getType)
+                .containsExactly(BidType.banner);
+    }
+
+    private static Imp givenImp(UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+        return impCustomizer.apply(Imp.builder()
+                        .id("123")
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpTaboola.builder()
+                                        .publisherId("token")
+                                        .publisherDomain("test.com")
+                                        .tagId("1")
+                                        .bidFloor(BigDecimal.TEN)
+                                        .build()))))
+                .build();
+    }
+
+    private static BidderCall<BidRequest> givenHttpCall(BidRequest bidRequest, String body) {
+        return BidderCall.succeededHttp(
+                HttpRequest.<BidRequest>builder().payload(bidRequest).build(),
+                HttpResponse.of(200, null, body),
+                null);
+    }
+
+    private static BidRequest givenBidRequest(
+            UnaryOperator<BidRequest.BidRequestBuilder> bidRequestCustomizer,
+            UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+
+        return bidRequestCustomizer.apply(BidRequest.builder()
+                        .imp(List.of(givenImp(impCustomizer))))
+                .build();
+    }
+
+    private static BidRequest givenBidRequest(UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+        return givenBidRequest(identity(), impCustomizer);
+    }
```

Comment:
What about to use this methods? It will allow you to easily use multiple imps in one test case.
```
    private static BidRequest givenBidRequest(UnaryOperator<BidRequest.BidRequestBuilder> bidRequestCustomizer,
                                              Imp... imps) {

        return bidRequestCustomizer.apply(BidRequest.builder().imp(List.of(imps))).build();
    }

    private static Imp givenImp(UnaryOperator<Imp.ImpBuilder> impCustomizer,
                                UnaryOperator<ExtImpTaboola.ExtImpTaboolaBuilder> extImpCustomizer) {

        return impCustomizer
                .apply(Imp.builder().ext(mapper.valueToTree(extImpCustomizer.apply(ExtImpTaboola.builder()).build())))
                .build();
    }

```